Realm/Realm/Sync/MongoClient.cs

```diff
@@ -44,12 +48,39 @@ public class MongoClient
         /// <value>The name of the remote MongoDB service.</value>
         public string ServiceName { get; }
 
+        private static readonly ConcurrentDictionary<Type, ObjectSchema> SchemaByName = new();
+
         internal MongoClient(User user, string serviceName)
         {
             User = user;
             ServiceName = serviceName;
         }
 
+        /// <summary>
+        /// Gets a collection of documents from MongoDB that can be deserialized in Realm objects.
+        /// </summary>
+        /// <remarks>
+        /// The collection and database name are automatically derived from the Realm object class.
+        /// </remarks>
+        /// <typeparam name="TRealmObject">The Realm object type that matches the shape of the documents in the collection.</typeparam>
+        /// <returns>A <see cref="Collection{TRealmObject}"/> instance that exposes an API for CRUD operations on its contents.</returns>
+        public Collection<TRealmObject> GetCollection<TRealmObject>()
+            where TRealmObject : class, IRealmObjectBase
+        {
+            if (!SchemaByName.TryGetValue(typeof(TRealmObject), out var schema))
+            {
+                throw new NotSupportedException("This method is only supported for source-generated classes - i.e. ones that inherit from IRealmObject rather than RealmObject.");
+            }
+
+            return new Collection<TRealmObject>(this, schema.Name);
+        }
+
+        //TODO This has been done to avoid the reflection in the GetCollection method. Need to look at this again
```

Comment:
I'm not very convinced by this. I think we talked about this at some point but I don't remember, could we use something similar to create the default schema for realm?